### PR TITLE
Improve support for M2E Pro

### PR DESCRIPTION
**Pull request by ryanreeves-taxjar**
When syncing orders/refunds to TaxJar, first check to see if M2ePro is
installed.  If so, try to load the original provider from the order.

### CODE REVIEW

File: Model/Transaction.php [hunk 1]
```diff
@@ -274,6 +283,32 @@ function buildCustomerExemption($order)
...
+
+        try {
+            if (class_exists('\Ess\M2ePro\Model\Order')) {
+                $m2eOrder = $this->objectManager->create('\Ess\M2ePro\Model\Order');
+                $m2eOrder = $m2eOrder->load($order->getId(), 'magento_order_id');
+            }
+
+            if (isset($m2eOrder)) {
+                if (in_array($m2eOrder->getMarketplace()->getComponentMode(), ['amazon', 'ebay', 'walmart'])) {
+                    $provider = $m2eOrder->getMarketplace()->getComponentMode();
```
Comment (ryanreeves-taxjar) [inline Model/Transaction.php]: Confirm getComponentMode() loads correctly.

File: Model/Transaction.php [hunk 2]
```diff
@@ -57,28 +57,36 @@ class Transaction
...
      */
     public function __construct(
         \Magento\Framework\App\Config\ScopeConfigInterface $scopeConfig,
         \Taxjar\SalesTax\Model\ClientFactory $clientFactory,
         \Magento\Catalog\Model\ProductRepository $productRepository,
         \Magento\Directory\Model\RegionFactory $regionFactory,
         \Magento\Tax\Api\TaxClassRepositoryInterface $taxClassRepository,
-        \Taxjar\SalesTax\Model\Logger $logger
+        \Taxjar\SalesTax\Model\Logger $logger,
+        \Magento\Framework\ObjectManagerInterface $objectmanager
```
Comment (fastdivision) [inline Model/Transaction.php]: @ryanreeves-taxjar Consider camel casing `$objectmanager` to `$objectManager`

File: Model/Transaction.php [hunk 3]
```diff
@@ -274,6 +282,30 @@ function buildCustomerExemption($order)
         return [];
     }
 
+    /**
+     * @param \Magento\Sales\Model\Order $order
+     * @return string
+     */
+    public function getProvider($order)
```
Comment (fastdivision) [inline Model/Transaction.php]: @ryanreeves-taxjar This function should have a `protected` scope.

Comment (ryanreeves-taxjar) [reply Model/Transaction.php]: I made this public when we were first looking at using a plugin around this function and forgot to change it back.

**Comment by fastdivision** [comment]
Tested locally, no issues passing default provider ID with M2E disabled 👍

**Pull request merged**